courses/utils.py

```diff
@@ -25,15 +26,26 @@ def ensure_course_run_grade(user, course_run, edx_grade, should_update=False):
 
     updated = False
     if should_update:
-        grade, created = CourseRunGrade.objects.update_or_create(
-            course_run=course_run, user=user, defaults=grade_properties
-        )
-        updated = not created
+        with transaction.atomic():
+            run_grade, created = CourseRunGrade.objects.select_for_update().get_or_create(
+                course_run=course_run,
+                user=user,
+                defaults=grade_properties
+            )
+
+            if not created and not run_grade.set_by_admin and run_grade.grade != edx_grade.percent:
```

Comment:
Can you please update this to use `has_equal_properties`? That would be more accurate. This only checks for the percentage grade and does not consider `passed` or `letter_grade` values.